php_phongo.c

```diff
@@ -1878,17 +1264,98 @@ static bool php_phongo_apply_wc_options_to_uri(mongoc_uri_t *uri, bson_t *option
 	return true;
 } /* }}} */
 
+static bool php_phongo_apply_ssl_opts(mongoc_client_t *client, zval *zdriverOptions TSRMLS_DC)
+{
+	mongoc_ssl_opt_t ssl_opt = {0};
+	zval *zoptions;
+	zend_bool free_pem_file = 0, free_pem_pwd = 0, free_ca_file = 0, free_ca_dir = 0;
+	int plen;
+
+	if (!zdriverOptions) {
+		return false;
+	}
+
+	zoptions = zdriverOptions;
+
+	if (php_array_existsc(zdriverOptions, "context")) {
+		php_stream_context *context;
+		zval *zcontext;
+
+		zcontext = php_array_fetchc(zdriverOptions, "context");
+
+		/* TODO: Do we want to check global stream options, too? */
+		context = php_stream_context_from_zval(zcontext, 1);
+
+		if (!context) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "\"context\" driver option is not a valid Stream-Context resource");
+			return false;
+		}
+
+#if PHP_VERSION_ID >= 70000
+		zoptions = php_array_fetchc_array(&context->options, "ssl");
+#else
+		zoptions = php_array_fetchc_array(context->options, "ssl");
+#endif
+
+		if (!zoptions) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Stream-Context resource does not contain \"ssl\" options array");
+			return false;
+		}
+	}
+
+	if (php_array_existsc(zoptions, "verify_hostname")) {
+		ssl_opt.allow_invalid_hostname = !php_array_fetchc_bool(zoptions, "verify_hostname");
```

Comment:
@derickr: HHVM doesn't appear to support this at all.

Also, PHP's own [SSL context options](http://php.net/manual/en/context.ssl.php) don't have anything called `verify_hostname`, so I propose we just name our option `allow_invalid_hostname`.